brewlette-frontend/src/App.js

```diff
@@ -0,0 +1,97 @@
+import Cafe from "./components/Cafe";
+import Logo from "./components/Logo";
+import CafeMenu from "./components/CafeMenu";
+import Search from "./components/Search";
+import { getCoordinates, getCafes, getCafeDetails } from "./api/api";
+import "./App.css";
+import { neighborhoods } from "./neighborhoods";
+import React, { useState, useEffect, useRef } from "react";
+import shuffle from "./helpers/helpers";
+
+function App() {
+  const [neighborhood, setNeighborhood] = useState("");
+  const [count, setCount] = useState(0);
+  const [coords, setCoords] = useState("");
+  const [cafes, setCafes] = useState([]);
+  const [cafe, setCafe] = useState(null);
```

Comment:
I think it can be helpful to think about what the complete data model needed for this app is. It seems to be something like:

```typescript
AppState = {
  neighborhoods: String[],
  neighborhood: String,
  coords: String,
  cafes: Cafe[],
  selectedCafe: CafeDetail,
  count: number;
}

// It seems like a few changes could be:

AppState = {
  neighborhoods: String[],
  neighborhood: {
    name: String,
    coords: {
      lat: number,
      lng: number
    },
    cafes: Cafe[],
    cafeIndex: number,
  },
  selectedCafe: CafeDetail,
}

// I think this is equivalent but captures the idea that the coordinates, cafes, and currently selected cafe index are properties of a neighborhood
```